### ISSUE TITLE
Latex regexes should ignore square bracket prefix

### ISSUE THREAD

**Issue by ilyaminati** [mangecoeur/document-outline#22]
Section headings like this don't show up in the outline:

    \section[Foo]{Bar baz}

Removing `[Foo]` allows the heading to be displayed.

**Comment by mangecoeur** [comment]
Will have a look, didn't actually know this was allowed syntax for a latex heading. What is the function of this syntax?

**Comment by ilyaminati** [comment]
It's standard Latex syntax in general. For example:

```
\documentclass[notitlepage]{nature}
\usepackage[roman]{parnotes}
\usepackage[unicode=false]{hyperref}
```

For section titles, the bit in square brackets is used for the table of contents. See https://tex.stackexchange.com/questions/309289/square-brackets-for-section-and-chapter-what-is-the-difference-between-secti

It'd actually be kinda cool to use the square brackets as the content of the outline.

Also, note this kind of command. Now, the outline says "Longer Title Beyond TOC\protect\footnote{This is a footnote}", but it should probably say "Longer Title Beyond TOC" or possibly "TOC Entry".

```
\chapter[TOC Entry]{Longer Title Beyond TOC\protect\footnote{This is a footnote}}
```

**Comment by mangecoeur** [comment]
Ok I will see what I can do with the heading regex if it can handle it without breaking the simpler cases (like when you have a `/` character in your normal title) -  I don't feel like building a full Latex parser in Javascript :P

**Comment by mangecoeur** [comment]
closed by 1.6.1
